lib/microsoft_kiota_serialization_json/json_serialization_writer.rb

```diff
@@ -152,16 +152,17 @@ def write_collection_of_object_values(key, values)
           end
         end
         @writer[key] = values.map do |v|
-          self.write_object_value(nil, v).writer
+          self.write_object_value(nil, v, true).writer
         end
       end
     end
 
-    def write_object_value(key, value)
```

Comment:
The problem is here and can also be addressed there if more appropriate:
https://github.com/microsoft/kiota-abstractions-ruby/blob/main/lib/microsoft_kiota_abstractions/request_information.rb#L99-L101

Since with both objects and arrays we don't modify the writer only return a temp version. We don't actually set anything there.

I was going by the assumption that if you pass nil as the first param you want to add the value to the root. If that is the wrong assumption the I can move my PR to `kiota-abstractions-ruby`

Reply:
the assumption is correct.
I think the issue is that having any of the "write" methods returning a new writer. This was a temporary workaround done by the intern years ago so his demo could work at the time, but it diverges from the other language implementations and this is what we should look to correct here.